pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -2510,8 +2510,13 @@ private int getMessageTTL() throws Exception {
     public CompletableFuture<Void> addSchemaIfIdleOrCheckCompatible(SchemaData schema) {
         return hasSchema()
             .thenCompose((hasSchema) -> {
-                if (hasSchema || isActive(InactiveTopicDeleteMode.delete_when_no_subscriptions)
-                        || ledger.getTotalSize() != 0) {
+                int numActiveConsumers = subscriptions.values().stream()
+                        .map(subscription -> subscription.getConsumers().size())
+                        .reduce(0, Integer::sum);
```

Comment:
Count ?

Reply:
If you mean the `numActiveConsumers`, yes.

Reply:
Sorry, this is a working example
```
 int numActiveConsumers = subscriptions.values().stream()
                        .mapToInt(subscription -> subscription.getConsumers().size())
                        .sum();
```

btw this is not a big deal, no need to change

Reply:
BTW, for non-persistent topics I'll fix it soon.